File: cpp/p2/soln.cpp
#include <iostream>

using namespace std;

int main() {
	int max_fib = 4000000;
	int accum = 0;
	int tmp = 0;
	int x = 0;
	int y = 1;

	while (y < max_fib){
		if (y % 2 == 0){
			accum += y;
		}
		tmp = y;
		y = x + y;
		x = tmp;
	}
	cout << accum << "\n";
}

File: R/p2/soln.R
max_fib <- 4000000
accum <- 0
x <- 0
y <- 1
while (y < max_fib){
	if (y %% 2 == 0){
		accum <- accum + y
	}
	tmp <- y
	y <- x + y
	x <- tmp
}

print(accum)
